dataService.js

```diff
@@ -0,0 +1,45 @@
+// import required json arrays
+const allCategories = require('./src/data/categories.json').categories
```

Comment:
Don't load this data in advance, load it data inside the getProducts method when productMap is null.